src/lib.rs

```diff
@@ -88,20 +90,29 @@ impl Contract {
         }
     }
 
-    pub fn add_like(&mut self, post_id: PostId) {
+    #[payable]
+    pub fn add_like(&mut self, post_id: PostId) -> Promise {
         near_sdk::log!("add_like");
         let mut post: Post = self
             .posts
             .get(post_id)
             .unwrap_or_else(|| panic!("Post id {} not found", post_id))
             .into();
+        let post_author = post.author_id.clone();
         let like =
             Like { author_id: env::predecessor_account_id(), timestamp: env::block_timestamp() };
         post.likes.insert(like);
         self.posts.replace(post_id, &post.into());
+        notify::notify_like(post_id, post_author)
     }
 
-    pub fn add_post(&mut self, parent_id: Option<PostId>, body: PostBody, labels: HashSet<String>) {
+    #[payable]
+    pub fn add_post(
+        &mut self,
+        parent_id: Option<PostId>,
+        body: PostBody,
+        labels: HashSet<String>,
+    ) -> Option<Promise> {
```

Comment:
I think https://docs.rs/near-sdk/latest/near_sdk/enum.PromiseOrValue.html is more correct here.

Reply:
Similar as above, since we don't need to wait for notify, I think dropping it upon creation will be the case, so this function will return unit. 

Regarding `PromiseOrValue<()>` vs `Option<Promise>`, I thought the later is fine because I observed it in non_fungible_token standard: https://github.com/near/near-sdk-rs/blob/770cbce018a1b6c49d58276a075ace3da96d6dc1/near-contract-standards/src/non_fungible_token/approval/approval_impl.rs#L29

Reply:
I think then NFT standard has a bug. I am not sure returning anything as a type besides `PromiseOrValue`, `Promise` or regular value works. CC @frol

Reply:
I think then NFT standard has a bug. I am not sure returning anything as a type besides `PromiseOrValue`, `Promise` or regular value works. CC @frol

Reply:
I also believe `PromiseOrValue` should be the return type. I don't know why `nft_approve` has this signature, but I can dive into it if necessary.